src/object_trajectory_predictor.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file
+ * @brief     This file contains the ObjectTrajectoryPredictor, it is the main class of the module ObjectPositionalPrediction.
+ *
+ * @author    Max Beunk & Daniel van Vliet
+ * @license   See LICENSE
+ */
+
+#ifndef OBJECTTRAJECTORYPREDICTOR_HPP
+#define OBJECTTRAJECTORYPREDICTOR_HPP
+
+#include "object_data.hpp"
+#include "vector3d.hpp"
+#include <array>
+
+class ObjectTrajectoryPredictor {
+  private:
+    std::array<ObjectData, 3> objectSampleData;
```

Comment:
You may want to make the ObjectTrajectoryPredictor object a template. In this way, users are able to provide their own desired array sizes.